src/wtmp.c

```diff
@@ -40,7 +40,7 @@ int log_wtmp(const char *line, const char *name, const char *host,
     !(defined(LINUX) || defined(__hpux) || defined (_AIX))
   /* This "auxilliary" utmp doesn't exist under linux. */
 
-#if defined(__sparcv9) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#if defined(__sun)
```

Comment:
Could you explain more how you mean it?
I can see how we can move definition of WTMPX_FILE into the `configure` script.
Do you mean there would be a specific macro that would tell us whether to use `struct utmpx` or `struct futmpx`?
I think that such a move should be done as a separate commit anyway, and this pull request should just fix this one bug.